data/update-scriptworker-revisions.py

```diff
@@ -46,17 +46,20 @@ async def main():
             if d["repo"] in r:
                 d["sha"] = r[d["repo"]]
 
+    # Read the file
+    with open(common_yaml, 'r') as f:
+        content = f.read()
+
+    # Apply replacements
     for directive in DIRECTIVES:
-        cmd = [
-            "sed",
-            "-i",  # in-place
-            "",
-            "-E",  # extended
-            f"s/(.*{directive['common_key']}\\: ).*/\\1\"{directive['sha']}\"/g",
-            common_yaml,
-        ]
-        print(f"Running command: {cmd}")
-        subprocess.run(cmd)
+        pattern = f"(.*{directive['common_key']}\\: ).*"
+        replacement = f"\\1\"{directive['sha']}\""
+        content = re.sub(pattern, replacement, content)
+        print(f"Replaced {directive['common_key']} with {directive['sha']}")
+
+    # Write the file back
+    with open(common_yaml, 'w') as f:
+        f.write(content)
```

Comment:
Would be best to write the file back atomically, i.e. write to common.yaml.tmp then os.rename it.